mkosi/__main__.py

```diff
@@ -3,23 +3,28 @@
 
 import contextlib
 import os
+import subprocess
 import sys
 from collections.abc import Iterator
-from subprocess import CalledProcessError
 
 from mkosi import load_args, run_verb
 from mkosi.config import MkosiConfigParser
-from mkosi.log import MkosiException, die
+from mkosi.log import MkosiPrinter, die, ARG_DEBUG
 from mkosi.run import excepthook
 
 
 @contextlib.contextmanager
 def propagate_failed_return() -> Iterator[None]:
     try:
         yield
-    except MkosiException as e:
+    except BaseException as e:
```

Comment:
This one is probably right for once, at least to some extent. `KeyboardInterrupt` (and possibly `SystemExit` for good measure) should be handleded separately, so this should should just catch `Exception` instead of `BaseException`